tests/testTriggers.go

```diff
@@ -7319,6 +7319,359 @@ func TriggerMetroDR(contexts *[]*scheduler.Context, recordChan *chan *EventRecor
 	updateMetrics(*event)
 }
 
+// TriggerAsyncDRPXRestartSource triggers Async DR with PX restart on source
+func TriggerAsyncDRPXRestartSource(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer endLongevityTest()
+	startLongevityTest(AsyncDRPXRestartSource)
+	defer ginkgo.GinkgoRecover()
+	log.Infof("Async DR PX restart on source trigger triggered at: %v", time.Now())
+	defer ginkgo.GinkgoRecover()
+	event := &EventRecord{
+			Event: Event{
+					ID:   GenerateUUID(),
+					Type: AsyncDRPXRestartSource,
+			},
+			Start:   time.Now().Format(time.RFC1123),
+			Outcome: []error{},
+	}
+	defer func() {
+			event.End = time.Now().Format(time.RFC1123)
+			*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	chaosLevel := ChaosMap[AsyncDRPXRestartSource]
+	var (
+			migrationNamespaces   []string
+			taskNamePrefix        = "async-dr-pxrs"
+			allMigrations         []*storkapi.Migration
+			includeVolumesFlag    = true
+			includeResourcesFlag  = true
+			startApplicationsFlag = false
+	)
+
+	Step(fmt.Sprintf("Deploy applications for migration, with frequency: %v", chaosLevel), func() {
+
+			// Write kubeconfig files after reading from the config maps created by torpedo deploy script
+			err := asyncdr.WriteKubeconfigToFiles()
+			if err != nil {
+					log.Errorf("Failed to write kubeconfig: %v", err)
+					UpdateOutcome(event, err)
+                    return
+			}
+
+			err = SetSourceKubeConfig()
+			if err != nil {
+					log.Errorf("Failed to Set source kubeconfig: %v", err)
+					UpdateOutcome(event, err)
+                    return
+			}
+			UpdateOutcome(event, err)
```

Comment:
remove this as you are handling it in L7366

Reply:
Removed.